composer.json

```diff
@@ -39,7 +39,8 @@
     "autoload-dev": {
         "psr-4": {
             "Tests\\": "tests/",
-            "App\\": "tests/laravel/app"
+            "App\\": "tests/laravel/app",
+            "Modules\\": "tests/laravel/modules"
```

Comment:
Can't we detect the `project_scan_dirs` and `project_namespaces` from this value?

Reply:
Oh yeah, that seems obvious now!  

For the benefit of the tests we would need to fetch mappings from `autoload` and `autoload-dev`.  Are you aware of an API exposed by composer to read the active project's composer file?  When running the tests in this package the path to composer.json will be different to when vendored into a laravel project.

Reply:
So i have a solution...
```
$composer = Composer\Factory::create(new Composer\IO\NullIO(), null,null);
$activeJsonPath = $composer->getConfig()->getConfigSource()->getName();
// $activeJsonPath == '/Users/mark/code/scout-extended/composer.json'
```

Reply:
I've refactored by interrogating Composer - this enhancement is now zero configuration!